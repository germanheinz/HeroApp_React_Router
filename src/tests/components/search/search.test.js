import '@testing-library/jest-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';
import { MemoryRouter, Route } from 'react-router';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing <SearchScreen />', () => {

    test('Debe mostrarse con los valores por defecto ', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );
        console.log(wrapper.html())
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero');
    })

    test('Debe de mostrar a Batman y el input con el valor del queryString', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );

        expect(wrapper.find('input').prop('value')).toBe('batman');



    })
    
})
