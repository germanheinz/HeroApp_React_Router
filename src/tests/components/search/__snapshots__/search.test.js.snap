// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchScreen /> Debe mostrarse con los valores por defecto  1`] = `ReactWrapper {}`;
